src/Storages/MergeTree/MergeTreePartInfo.h

```diff
@@ -191,4 +191,21 @@ struct DetachedPartInfo : public MergeTreePartInfo
 
 using DetachedPartsInfo = std::vector<DetachedPartInfo>;
 
+using PartitionIds = std::vector<String>;
```

Comment:
Please add a comment, explaining that most of the code assumes `PartitionIds` to be sorted and suitable for binary search.